profile-models/src/main/java/com/telecominfraproject/wlan/profile/rf/models/RfElementConfiguration.java

```diff
@@ -0,0 +1,413 @@
+//		this.perimeterDetectionEnabled = perimeterDetectionEnabled;
+//	}
+//
+//	public ChannelHopSettings getChannelHopSettings() {
+//		return channelHopSettings;
+//	}
+//
+//	public void setChannelHopSettings(ChannelHopSettings channelHopSettings) {
+//		this.channelHopSettings = channelHopSettings;
+//	}
+//
+//	public RadioBestApSettings getBestApSettings() {
+//		return bestApSettings;
+//	}
+//
+//	public void setBestApSettings(RadioBestApSettings bestApSettings) {
+//		this.bestApSettings = bestApSettings;
+//	}
+
+	@Override
+    public RfElementConfiguration clone() {
+        return (RfElementConfiguration) super.clone();
+    }
+	
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ((activeScanSettings == null) ? 0 : activeScanSettings.hashCode());
+		result = prime * result + (autoChannelSelection ? 1231 : 1237);
+		result = prime * result + ((beaconInterval == null) ? 0 : beaconInterval.hashCode());
+//		result = prime * result + ((bestApEnabled == null) ? 0 : bestApEnabled.hashCode());
+//		result = prime * result + ((bestApSettings == null) ? 0 : bestApSettings.hashCode());
+		result = prime * result + ((channelBandwidth == null) ? 0 : channelBandwidth.hashCode());
+//		result = prime * result + ((channelHopSettings == null) ? 0 : channelHopSettings.hashCode());
+//		result = prime * result + ((clientDisconnectThresholdDb == null) ? 0 : clientDisconnectThresholdDb.hashCode());
+//		result = prime * result + ((eirpTxPower == null) ? 0 : eirpTxPower.hashCode());
+		result = prime * result + ((forceScanDuringVoice == null) ? 0 : forceScanDuringVoice.hashCode());
+//		result = prime * result + ((managementRate == null) ? 0 : managementRate.hashCode());
+		result = prime * result + ((maxNumClients == null) ? 0 : maxNumClients.hashCode());
+		result = prime * result + ((mimoMode == null) ? 0 : mimoMode.hashCode());
+//		result = prime * result + ((minAutoCellSize == null) ? 0 : minAutoCellSize.hashCode());
+		result = prime * result + ((multicastRate == null) ? 0 : multicastRate.hashCode());
+//		result = prime * result + ((neighbouringListApConfig == null) ? 0 : neighbouringListApConfig.hashCode());
+//		result = prime * result + ((perimeterDetectionEnabled == null) ? 0 : perimeterDetectionEnabled.hashCode());
+//		result = prime * result + ((probeResponseThresholdDb == null) ? 0 : probeResponseThresholdDb.hashCode());
+		result = prime * result + ((rf == null) ? 0 : rf.hashCode());
+		result = prime * result + ((rtsCtsThreshold == null) ? 0 : rtsCtsThreshold.hashCode());
+//		result = prime * result + ((rxCellSizeDb == null) ? 0 : rxCellSizeDb.hashCode());
+		return result;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		RfElementConfiguration other = (RfElementConfiguration) obj;
+		if (activeScanSettings == null) {
```

Comment:
Can use Objects.equals() for each member object to avoid all the null checks

Reply:
Same comment as hashCode